src/error.rs

```diff
@@ -0,0 +1,27 @@
+use crate::consts::*;
+use core::fmt::{Display, Formatter};
+
+pub type Result<T, E = Error> = core::result::Result<T, E>;
+
+#[derive(Debug)]
+pub enum Error {
```

Comment:
This error type should belong to `clock.rs` and not the whole crate. Otherwise you end up in a situation when you need to process the `PllInputFrequencyTooLow` error during UART write operation, for example.